File: base/src/main/java/io/quarkus/code/rest/exceptions/IllegalArgumentExceptionMapper.java
package io.quarkus.code.rest.exceptions;

import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.ext.ExceptionMapper;
import jakarta.ws.rs.ext.Provider;

@Provider
public class IllegalArgumentExceptionMapper implements ExceptionMapper<IllegalArgumentException> {
    @Override
    public Response toResponse(IllegalArgumentException e) {
        String message = "Bad request > " + e.getMessage();
        return Response.status(Response.Status.BAD_REQUEST)
                .entity(message)
                .type(MediaType.TEXT_PLAIN)
                .build();
    }
}

File: api/src/main/kotlin/io/quarkus/code/rest/IllegalArgumentExceptionMapper.kt
package io.quarkus.code.rest

import jakarta.ws.rs.core.MediaType.TEXT_PLAIN
import jakarta.ws.rs.core.Response
import jakarta.ws.rs.ext.ExceptionMapper
import jakarta.ws.rs.ext.Provider

@Provider
class IllegalArgumentExceptionMapper: ExceptionMapper<IllegalArgumentException>  {
    override fun toResponse(e: IllegalArgumentException): Response {
        val message = "Bad request > ${e.message}"
        return Response.status(Response.Status.BAD_REQUEST)
            .entity(message)
            .type(TEXT_PLAIN)
            .build()
    }
}